core/src/main/java/com/flipkart/poseidon/legoset/PoseidonLegoSet.java

```diff
@@ -283,19 +290,26 @@ public void setContext(ApplicationContext context) {
 
     private void resolveInjectableConstructorDependencies(Constructor<?> constructor, Object[] initParams, int offset, Optional<Request> request) throws MissingInformationException, ElementNotFoundException {
         for (int i = offset; i < constructor.getParameterCount(); i++) {
-            final RequestAttribute requestAttribute = constructor.getParameters()[i].getAnnotation(RequestAttribute.class);
-            final com.flipkart.poseidon.datasources.ServiceClient serviceClientAttribute = constructor.getParameters()[i].getAnnotation(com.flipkart.poseidon.datasources.ServiceClient.class);
+            final Parameter constructorParameter = constructor.getParameters()[i];
+            final RequestAttribute requestAttribute = constructorParameter.getAnnotation(RequestAttribute.class);
+            final com.flipkart.poseidon.datasources.ServiceClient serviceClientAttribute = constructorParameter.getAnnotation(com.flipkart.poseidon.datasources.ServiceClient.class);
             if (requestAttribute != null) {
-                final String attributeName = StringUtils.isNullOrEmpty(requestAttribute.value()) ? constructor.getParameters()[i].getName() : requestAttribute.value();
-                initParams[i] = request.map(r -> r.getAttribute(attributeName)).orElse(null);
+                final String attributeName = StringUtils.isNullOrEmpty(requestAttribute.value()) ? constructorParameter.getName() : requestAttribute.value();
+                initParams[i] = request.map(r -> r.getAttribute(attributeName)).map(attr -> {
+                    if (constructorParameter.getType().isAssignableFrom(attr.getClass())) {
```

Comment:
NPE possible in attr when attribute isn't present in request?

Reply:
i'm using Optional.map, this should only be executed when attr is present

Reply:
Oops, overlooked it.